lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -476,101 +539,7 @@ class MouseAdapter extends BaseAdapter {
-/// splitting it into two integer components: milliseconds + microseconds.
-Duration _eventTimeStampToDuration(num milliseconds) {
-  final int ms = milliseconds.toInt();
-  final int micro =
-      ((milliseconds - ms) * Duration.microsecondsPerMillisecond).toInt();
-  return Duration(milliseconds: ms, microseconds: micro);
-}
-
-void _ensureMouseDeviceAdded(List<ui.PointerData> data, double clientX,
-    double clientY, int buttons, double timeStamp, int deviceId) {
-  if (PointerBinding.instance._activePointerIds.contains(deviceId)) {
-    return;
-  }
-  PointerBinding.instance._activePointerIds.add(deviceId);
-  // Only send [PointerChange.add] the first time.
-  data.insert(
-      0,
-      ui.PointerData(
-        change: ui.PointerChange.add,
-        timeStamp: _eventTimeStampToDuration(timeStamp),
-        kind: ui.PointerDeviceKind.mouse,
-        // In order for Flutter to actually add this pointer, we need to set the
-        // signal to none.
-        signalKind: ui.PointerSignalKind.none,
-        device: deviceId,
-        physicalX: clientX * ui.window.devicePixelRatio,
-        physicalY: clientY * ui.window.devicePixelRatio,
-        buttons: buttons,
-        pressure: 1.0,
-        pressureMin: 0.0,
-        pressureMax: 1.0,
-        scrollDeltaX: 0,
-        scrollDeltaY: 0,
-      ));
-}
-
-List<ui.PointerData> _convertWheelEventToPointerData(
-  html.WheelEvent event,
-) {
-  const int domDeltaPixel = 0x00;
-  const int domDeltaLine = 0x01;
-  const int domDeltaPage = 0x02;
-
-  // Flutter only supports pixel scroll delta. Convert deltaMode values
-  // to pixels.
-  double deltaX = event.deltaX;
-  double deltaY = event.deltaY;
-  switch (event.deltaMode) {
-    case domDeltaLine:
-      deltaX *= 32.0;
-      deltaY *= 32.0;
-      break;
-    case domDeltaPage:
-      deltaX *= ui.window.physicalSize.width;
-      deltaY *= ui.window.physicalSize.height;
-      break;
-    case domDeltaPixel:
-    default:
-      break;
-  }
-
-  final List<ui.PointerData> data = <ui.PointerData>[];
-  _ensureMouseDeviceAdded(data, event.client.x, event.client.y, event.buttons,
-      event.timeStamp, _mouseDeviceId);
-  data.add(ui.PointerData(
-    change: ui.PointerChange.hover,
-    timeStamp: _eventTimeStampToDuration(event.timeStamp),
-    kind: ui.PointerDeviceKind.mouse,
-    signalKind: ui.PointerSignalKind.scroll,
-    device: _mouseDeviceId,
-    physicalX: event.client.x * ui.window.devicePixelRatio,
-    physicalY: event.client.y * ui.window.devicePixelRatio,
-    buttons: event.buttons,
-    pressure: 1.0,
-    pressureMin: 0.0,
-    pressureMax: 1.0,
-    scrollDeltaX: deltaX,
-    scrollDeltaY: deltaY,
-  ));
-  return data;
-}
-
-void _addWheelEventListener(void listener(html.WheelEvent e)) {
-  final dynamic eventOptions = js_util.newObject();
-  js_util.setProperty(eventOptions, 'passive', false);
-  js_util.callMethod(PointerBinding.instance.domRenderer.glassPaneElement,
```

Comment:
This is a bit confusing to me. If we will call preventDefault, why don't we just use capture = true, and then we can just use the more generic function BaseAdapter. _addEventListener?

Reply:
I'm not sure I understand.

The browser by default sets `passive=true` on `scroll` and `wheel` event listeners. So we need to explicitly set it to `passive=false` to be able to call `event.preventDefault()`. This is independent from `capture`.

cc @ferhatb who implemented wheel events.

Reply:
If we use usecapture= true, we don't have to call event.preventDefault()?

Reply:
Calling `event.preventDefault()` is necessary regardless of the value of `useCapture`.

`useCapture` controls when your listener gets called. If it's true, your listener is called before your children. If it's false, your children are called before your listener.

`event.preventDefault` tells the browser to not perform the default behavior for that event (which in this case is scrolling). We want Flutter to handle the scrolling, so we prevent the browser from doing any kind of scrolling.

Reply:
I just check we can still call preventdefault if we have usecapture = true, and from the api documentation of usecapture I can't see why we dont want to use it.

The reason why I want to move away from js_util.callMethod is that i am having trouble remove the listener afterward.

I updated it so that add wheel listener goes through BaseAdapter._addEventListener as well. Let me know if I missed something.

Reply:
offline discussion: we have to use js interop to remove the wheel event listener (`js_util.callMethod(glassPane, 'removeEventListener', ...)`).

Reply:
fixed